packages/cli/src/databases/repositories/workflow.repository.ts

```diff
@@ -99,30 +121,186 @@ export class WorkflowRepository extends Repository<WorkflowEntity> {
 			.execute();
 	}
 
-	async getMany(sharedWorkflowIds: string[], options: ListQuery.Options = {}) {
+	private buildBaseUnionQuery(workflowIds: string[], options: ListQuery.Options = {}) {
+		const subQueryParameters: ListQuery.Options = {
+			select: {
+				createdAt: true,
+				updatedAt: true,
+				id: true,
+				name: true,
+			},
+			filter: options.filter,
+		};
+
+		const columnNames = [...Object.keys(subQueryParameters.select ?? {}), 'resource'];
+
+		const [sortByColumn, sortByDirection] = this.parseSortingParams(
+			options.sortBy ?? 'updatedAt:asc',
+		);
+
+		const foldersQuery = this.folderRepository
+			.getManyQuery(subQueryParameters)
+			.addSelect("'folder'", 'resource');
+
+		const workflowsQuery = this.getManyQuery(workflowIds, subQueryParameters).addSelect(
+			"'workflow'",
+			'resource',
+		);
+
+		const qb = this.manager.createQueryBuilder();
+
+		return {
+			baseQuery: qb
+				.createQueryBuilder()
+				.addCommonTableExpression(foldersQuery, 'FOLDERS_QUERY', { columnNames })
+				.addCommonTableExpression(workflowsQuery, 'WORKFLOWS_QUERY', { columnNames })
+				.addCommonTableExpression(
+					`SELECT * FROM ${qb.escape('FOLDERS_QUERY')} UNION ALL SELECT * FROM ${qb.escape('WORKFLOWS_QUERY')}`,
+					'RESULT_QUERY',
+				),
+			sortByColumn,
+			sortByDirection,
+		};
+	}
+
+	async getWorkflowsAndFoldersUnion(workflowIds: string[], options: ListQuery.Options = {}) {
+		const { baseQuery, sortByColumn, sortByDirection } = this.buildBaseUnionQuery(
+			workflowIds,
+			options,
+		);
+
+		const query = baseQuery
+			.select(`${baseQuery.escape('RESULT')}.*`)
+			.from('RESULT_QUERY', 'RESULT');
+		if (sortByColumn === 'name') {
+			query
+				.addSelect(`LOWER(${baseQuery.escape('RESULT')}.${baseQuery.escape('name')})`, 'name_lower')
+				.orderBy('name_lower', sortByDirection);
```

Comment:
Selecting this is not necessary, this could be done in ORDER BY directly:

```sql
select * from workflow_entity
order by lower(name)
;
```

Could simplify things

Reply:
Thought the same, but that does not work on Sqlite.

https://michaelsoolee.com/case-insensitive-sorting-sqlite/